src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,45 +1,99 @@
-import { useState } from 'react';
+import { FormEvent, useState } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
 
-export const NewMovie = () => {
-  // Increase the count after successful form submission
-  // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+type Props = {
+  onAdd: (movie: Movie) => void
+};
+
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
+  const [count, setCount] = useState(0);
+  const [title, setTitle] = useState('');
+  const [imgUrl, setImgUrl] = useState('');
+  const [imdbUrl, setImdbUrl] = useState('');
+  const [imdbId, setImdbId] = useState('');
+  const [description, setDescription] = useState('');
+
+  // eslint-disable-next-line max-len
+  const pattern = /^((([A-Za-z]{3,9}:(?:\/\/)?)(?:[-;:&=+$,\w]+@)?[A-Za-z0-9.-]+|(?:www\.|[-;:&=+$,\w]+@)[A-Za-z0-9.-]+)((?:\/[+~%/.\w-_]*)?\??(?:[-+=&;%@.\w_]*)#?(?:[.!/\\\w]*))?)$/;
+
+  const addedMovie = {
+    title,
+    description,
+    imgUrl,
+    imdbUrl,
+    imdbId,
+  };
+
+  const clearForm = () => {
+    setTitle('');
+    setDescription('');
+    setImgUrl('');
+    setImdbUrl('');
+    setImdbId('');
+  };
+
+  const disabled = !title || !pattern.test(imgUrl)
+   || !pattern.test(imdbUrl) || !imdbId;
+
+  const handleSubmit = (event: FormEvent) => {
+    event.preventDefault();
+
+    if (!disabled) {
+      onAdd(addedMovie);
+    }
+
+    setCount(prevValue => prevValue + 1);
+    clearForm();
```

Comment:
Не потрібно очищувати форму якщо в ній є помилка

```suggestion
    if (disabled) {
      return
    }
    
    onAdd(addedMovie);

    setCount(prevValue => prevValue + 1);
    clearForm();
```